src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -1043,6 +1050,77 @@ internal static List<AzureSqlSkuCategory> GetEligibleSkuCategories(string target
+        /// </summary>
+        /// <param name="parameters">Parameters for the operation, as register during the type definition</param>
+        /// <param name="requestContext">Context provided by the framework</param>
+        /// <returns></returns>
+        internal async Task HandleCertificateMigrationRequest(
+          CertificateMigrationParams parameters,
+          RequestContext<CertificateMigrationResult> requestContext)
+        {
+            var result = new CertificateMigrationResult();
+
+            var credentials = new StaticTokenCredential(parameters.AccessToken); //New token provided, will change to shared ADS cache later.
+
+            // Reuse the tde migration client
+            var tdeMigrationClient = new TdeMigration(
+                   parameters.SourceSqlConnectionString,
+                   parameters.TargetSubscriptionId,
+                   parameters.TargetResourceGroupName,
+                   parameters.TargetManagedInstanceName,
+                   parameters.NetworkSharePath,
+                   parameters.NetworkShareDomain,
+                   parameters.NetworkShareUserName,
+                   parameters.NetworkSharePassword,
+                   credentials
+                   );
+
+            foreach (var dbName in parameters.EncryptedDatabases)
+            {
+                var migrationResult = await MigrateCertificate(tdeMigrationClient, dbName);
+                
+                var eventData = new CertificateMigrationProgressParams
+                {
+                    Name = dbName,
+                    Success = migrationResult.Success,
+                    Error = migrationResult.ErrorMessage
+                };
+                await requestContext.SendEvent(CertificateMigrationProgressEvent.Type, eventData);
+
+                result.MigrationStatuses.Add(migrationResult);
+            }
+
+            await requestContext.SendResult(result);
+        }
+
+        /// <summary>
+        /// Individual certificate migration operation
+        /// </summary>
+        /// <param name="tdeMigrationClient">Instance of the migration client</param>
+        /// <param name="dbName">Name of the database to migrate</param>
+        /// <returns></returns>
+        private async Task<CertificateMigrationEntryResult> MigrateCertificate(TdeMigration tdeMigrationClient,string dbName)
+        {
+            try
+            {
+                var result = await tdeMigrationClient.MigrateTdeCertificate(dbName, CancellationToken.None);
+
+                if (result is TdeExceptionResult tdeExceptionResult)
+                {
+                    return new CertificateMigrationEntryResult { DbName = dbName, Success = result.IsSuccess, ErrorMessage = tdeExceptionResult.Exception.Message };
+                }
+                else {
+                    return new CertificateMigrationEntryResult { DbName = dbName, Success = result.IsSuccess, ErrorMessage = result.UserFriendlyMessage };
```

Comment:
param name ErrorMessage doesn't make sense for this case

Reply:
Renamed all to Message